watcher/watcher.go

```diff
@@ -0,0 +1,156 @@
+package watcher
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Tonkpils/snag/exchange"
+	fsn "gopkg.in/fsnotify.v1"
+)
+
+const rebuild = "rebuild"
+
+type Watcher interface {
+	Watch(string) error
+}
+
+type FSWatcher struct {
+	ex           exchange.SendListener
+	fsn          *fsn.Watcher
+	done         chan struct{}
+	mtimes       map[string]time.Time
+	watching     map[string]struct{}
+	watchDir     string
+	ignoredItems []string
+}
+
+func New(ex exchange.SendListener, ignoredItems []string) (Watcher, error) {
```

Comment:
No, it actually would remove the dependency has on the `exchange`. It would place the logic of sending events to the exchange in the creator of it (`main` in this case). The exchange would still need to be notified just not implicitly by the watcher. Remember, the UI would only simply listen on events from the exchange and the builder would send the status of the build to the exchange. In this case for example, the watcher needs to know the event it's sending to even though the the caller defined it. If we changed the event we are subscribing to on the caller, that would mean a change in the watcher which means there's something up. Think if the watcher wasn't our package but 3rd party.